.github/workflows/pre-merge-integration-test.yml

```diff
@@ -0,0 +1,70 @@
+name: Pre-merge integration test
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - ready_for_review
+      - synchronize
+
+jobs:
+  deploy:
+    name: pre-merge-integration-tests
+    runs-on: ubuntu-latest
+    environment: di-ipv-cri-dev
+    timeout-minutes: 15
+    env:
+      AWS_REGION: eu-west-2
+      ENVIRONMENT: dev
+      STACK_NAME_PREFIX: pre-merge-integration
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: 11
+          distribution: zulu
+
+      - name: Setup SAM
+        uses: aws-actions/setup-sam@v1
+
+      - name: Assume temporary AWS role
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.GH_ACTIONS_ROLE_ARN }}
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: Gradle build
+        run: ./gradlew build -x test -x spotlessCheck
+
+      - name: SAM build
+        run: sam build -t infrastructure/lambda/template.yaml
+
+      - name: Set short SHA
+        id: vars
+        run: echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
+
+      - name: SAM deploy integration test stack
+        run: |
+          STACK_NAME=${{ env.STACK_NAME_PREFIX }}-${{ steps.vars.outputs.sha_short }}
+          sam deploy -t infrastructure/lambda/template.yaml \
+            --no-fail-on-empty-changeset \
+            --no-confirm-changeset \
+            --parameter-overrides EnvironmentName=${{ env.ENVIRONMENT }} \
+            --stack-name $STACK_NAME \
+            --s3-bucket ${{ secrets.ARTIFACT_SOURCE_BUCKET_NAME }} \
+            --s3-prefix $STACK_NAME \
+            --region ${{ env.AWS_REGION }} \
+            --capabilities CAPABILITY_IAM
+
+      - name: Run API integration tests
+        run: echo "🤞 now run integration tests..."
+
+      - name: Delete integration test stack
+        if: always()
```

Comment:
Do we need this line?

Reply:
we need it to always tear down the stack